### PR TITLE
fix: loading of duckdb every query

### PR DESCRIPTION
**Pull request by ozkatz**
Fix the duckDB wasm package being downloaded on every query (as well as a few minor tweaks to the UX of running said queries)

### CODE REVIEW

File: webui/src/pages/repositories/repository/fileRenderers/data.tsx [hunk 1]
```diff
@@ -26,24 +28,52 @@ LIMIT 20`
...
+        setLoading(true)
         withConnection(async conn => {
-            const results = await conn.query(query)
+            if (!sql || !sql.current) return
+            const results = await conn.query(sql.current.value)
             const data = results.toArray()
             setData(data)
             setError(null)
+
         }).catch(e => {
```
Comment (eladlachmi) [inline webui/src/pages/repositories/repository/fileRenderers/data.tsx:53]: You're mixing paradigms. `.catch` is Promise syntax and `async-await` is a different approach. If you're using `async-await`, you should use plain old `try-catch-finally`. That's the idea of `async-await` - writing sync-style code, and abstracting away all of the Promise handling, continuation, error handling, etc.

File: webui/src/pages/repositories/repository/fileRenderers/data.tsx [hunk 2]
```diff
@@ -26,24 +28,52 @@ LIMIT 20`
...
             setError(e.toString())
+        }).finally(() => {
+            setLoading(false)
         })
     }, [repoId, refId, path, shouldSubmit])
 
+
     let content;
+    let button = (
```
Comment (eladlachmi) [inline webui/src/pages/repositories/repository/fileRenderers/data.tsx:62]: IMO something along these lines would be cleaner and more readable.
```suggestion
    const button = (
      <Button type="submit" variant="success" disabled={loading}>
        <DatabaseIcon /> {" "}
        { loading ? "Executing..." : "Execute" }
      </Button>
    );
```

File: webui/src/pages/repositories/repository/fileRenderers/data.tsx [hunk 3]
```diff
@@ -58,59 +88,54 @@ LIMIT 20`
...
     }
 
-
     return (
         <div>
-            <Form onSubmit={(e) => {
-                e.preventDefault()
-                setShouldSubmit(!shouldSubmit)
-            }}>
+            <Form onSubmit={handleSubmit}>
```
Comment (eladlachmi) [inline webui/src/pages/repositories/repository/fileRenderers/data.tsx:126]: Like! 🤩

File: webui/src/pages/repositories/repository/fileRenderers/data.tsx [hunk 4]
```diff
@@ -58,59 +88,54 @@ LIMIT 20`
...
                 <Form.Group className="mb-2 mt-2" controlId="objectQuery">
-                    <Form.Control as="textarea" className="object-viewer-sql-input" rows={5} value={query} spellCheck={false} onChange={e => {
-                        setQuery(e.target.value)
-                    }} />
+                    <Form.Control as="textarea" className="object-viewer-sql-input" rows={5} defaultValue={initialQuery} spellCheck={false} ref={sql} />
```
Comment (eladlachmi) [inline webui/src/pages/repositories/repository/fileRenderers/data.tsx:128]: Not sure if it's something you want, but maybe add `autocomplete={false}`, so the native browser autocomplete thing doesn't pop up (unless you want. that, of course)

File: webui/src/pages/repositories/repository/fileRenderers/data.tsx [hunk 5]
```diff
@@ -58,59 +88,54 @@ LIMIT 20`
...
+
                 </Form.Group>
-                <Button type="submit" variant={"success"} >
-                    <DatabaseIcon/>{' '}
-                    Execute
-                </Button>
+                {button}
             </Form>
-            <div className={"mt-3 object-viewer-sql-results"}>
+            <div className={"mt-3"}>
```
Comment (eladlachmi) [inline webui/src/pages/repositories/repository/fileRenderers/data.tsx:138]: Not sure there's a need for the curly braces

**Review by eladlachmi — CHANGES_REQUESTED** [review]
Looks good!
Just a few relatively minor comments.

File: webui/src/pages/repositories/repository/fileRenderers/data.tsx (hunk 1 above)
Comment (ozkatz) [reply webui/src/pages/repositories/repository/fileRenderers/data.tsx]: Fixed

File: webui/src/pages/repositories/repository/fileRenderers/data.tsx (hunk 2 above)
Comment (ozkatz) [reply webui/src/pages/repositories/repository/fileRenderers/data.tsx]: Nice! used.

File: webui/src/pages/repositories/repository/fileRenderers/data.tsx (hunk 4 above)
Comment (ozkatz) [reply webui/src/pages/repositories/repository/fileRenderers/data.tsx]: Added!

**Review by eladlachmi — APPROVED** [review]
LGTM

**Pull request merged**